runtime/lua/vim/treesitter.lua

```diff
@@ -438,10 +426,11 @@ end
 ---@param lang (string|nil) Language of the parser (default: from buffer filetype)
 function M.start(bufnr, lang)
   bufnr = bufnr or api.nvim_get_current_buf()
-  local parser = M._get_parser(bufnr, lang)
+  local parser = M.get_parser(bufnr, lang, { error = false })
```

Comment:
use `assert` here?

Reply:
Later `if not parser then` gives a more informative error message.

Reply:
No, I mean use `assert` _here_ with the more informative error message. It's equivalent, of course; it's just a question of which pattern we want to promote `if not foo then error('bar')` vs. `assert(foo, 'bar')`).